controllers/watermarkpodautoscaler_controller.go

```diff
@@ -389,17 +389,36 @@ func shouldScale(logger logr.Logger, wpa *datadoghqv1alpha1.WatermarkPodAutoscal
 		transitionCountdown.With(prometheus.Labels{wpaNamePromLabel: wpa.Name, wpaNamespacePromLabel: wpa.Namespace, transitionPromLabel: "upscale", resourceNamespacePromLabel: wpa.Namespace, resourceNamePromLabel: wpa.Spec.ScaleTargetRef.Name, resourceKindPromLabel: wpa.Spec.ScaleTargetRef.Kind}).Set(0)
 	}
 
-	return canScale(logger, backoffUp, backoffDown, currentReplicas, desiredReplicas)
+	logger.Info("Cooldown status", "backoffUp", backoffUp, "backoffDown", backoffDown, "desiredReplicas", desiredReplicas, "currentReplicas", currentReplicas)
+
+	hasBeenAbove, aboveSince, err := getCondition(&wpa.Status, datadoghqv1alpha1.WatermarkPodAutoscalerStatusAboveHighWatermark)
+	if err != nil {
+		logger.Info("Could not retrieve condition about the time above Watermark, blocking potential scaling event", "error", err)
+		hasBeenAbove = corev1.ConditionFalse
+	}
+	if desiredReplicas > currentReplicas {
+		return canScale(logger, backoffUp, hasBeenAbove, aboveSince, wpa.Spec.UpscaleEvaluateAboveWatermarkSeconds)
+	}
+
+	hasBeenBelow, belowSince, err := getCondition(&wpa.Status, datadoghqv1alpha1.WatermarkPodAutoscalerStatusBelowLowWatermark)
+	if err != nil {
+		logger.Info("Could not retrieve condition about the time above Watermark, blocking potential scaling event", "error", err)
+		hasBeenBelow = corev1.ConditionFalse
+	}
+	if desiredReplicas < currentReplicas {
+		return canScale(logger, backoffDown, hasBeenBelow, belowSince, wpa.Spec.DownscaleEvaluateBelowWatermarkSeconds)
```

Comment:
why do we need `hasBeenBelow` since we already have `belowSince` 
`if belowSince > 0` it means `hasBeenBelow == true`

Reply:
just for safety, at the moment it's called when we know that we are out of the bounds but the timestamp is not updated.